xla/mlir/tools/mlir_interpreter/dialects/symbol_finder_linux.cc

```diff
@@ -0,0 +1,19 @@
+/* This code defines a POSIX-compatible function, FindSymbolInProcess, 
+   which uses dlsym to locate a function or variable symbol (symbol_name) in the current process;
+   it returns the symbol's address if found or an error if not */
+   
+#ifndef _WIN32
```

Comment:
Can you remove this `#ifndef _WIN32`? The whole file is Linux-only code and this is not needed.

Reply:
if I remove ifndef _WIN32, I get the following errors
xla/mlir/tools/mlir_interpreter/dialects/symbol_finder_linux.cc(7,10): fatal error: 'dlfcn.h' file not found
    7 | #include <dlfcn.h>
      |          ^~~~~~~~~
      
      xla/mlir/tools/mlir_interpreter/dialects/symbol_finder_linux.cc(14,21): error: use of undeclared identifier 'RTLD_DEFAULT'
   14 |   void* sym = dlsym(RTLD_DEFAULT, symbol_name.c_str());
      |                     ^

Reply:
What is your exact bazel test command?

This is supposed to work the following way:

1. Both `symbol_finder_linux` and `symbol_finder_windows` are tagged `manual`. That means they are not getting built by a wildcard build like `bazel build //xla/...`.
2. They are only getting built when you either specify the target explicitly on the command line (i.e. `bazel build //xla/mlir/tools/mlir_interpreter/dialects:symbol_finder_windows`) or when some other target depends on it.
3. `:symbol_finder` should always depend on the right implementation target, so there should be no error coming from there.

Reply:
> If I remove ifndef _WIN32, I get the following errors while running the bazel test on Windows
> 
> xla/mlir/tools/mlir_interpreter/dialects/symbol_finder_linux.cc(7,10): fatal error: 'dlfcn.h' file not found 7 | #include <dlfcn.h> | ^~~~~~~~~
> 
> ```
>   xla/mlir/tools/mlir_interpreter/dialects/symbol_finder_linux.cc(14,21): error: use of undeclared identifier 'RTLD_DEFAULT'
> ```
> 
> 14 | void* sym = dlsym(RTLD_DEFAULT, symbol_name.c_str()); | ^

Oh I see why this is not working. There is a `glob(*)` in the `:dialects` target.

Can you remove the glob there and list all the files explicitly? Then you should be able to remove the ifdefs.